h/static/scripts/directive/group-list.js

```diff
@@ -1,17 +1,64 @@
 'use strict';
 
+var assert = require('assert');
+
+function GroupsListController($scope) {
+  $scope.expandedGroupId = undefined;
+
+  // used to keep the dropdown from closing when the user
+  // interacts with the inline link share pane within the groups list
+  $scope.stopClickPropagation = function($event) {
+    $event.stopPropagation();
+  }
+
+  // show the share link for the specified group or clear it if
+  // null
+  $scope.toggleShareLink = function (groupId) {
+    if (!groupId || $scope.expandedGroupId === groupId) {
+      $scope.expandedGroupId = undefined;
+    } else {
+      $scope.expandedGroupId = groupId;
+    }
+  };
+
+  $scope.shouldShowShareLink = function (groupId) {
+    return $scope.expandedGroupId === groupId;
+  }
+
+  $scope.linkForGroup = function (groupId) {
+    return $scope.baseURI + 'groups/' + groupId;
+  }
+}
+
 /**
  * @ngdoc directive
  * @name groupList
  * @restrict AE
  * @description Displays a list of groups of which the user is a member.
  */
-
 // @ngInject
 module.exports = function (groups) {
   return {
-    link: function (scope, elem, attrs) {
-      scope.groups = groups;
+    controller: ['$scope', GroupsListController],
+    link: function ($scope, elem, attrs) {
+      $scope.groups = groups;
+
+      // set the base URI used later to construct the sharing
+      // link for the group
+      $scope.baseURI = elem[0].ownerDocument.baseURI;
```

Comment:
You shouldn't need to construct group URLs. The data sent from the server for groups [includes a prerendered URL](https://github.com/hypothesis/h/blob/4a324db9c3495e71a4a1c3752617d7b9e957e195/h/session.py#L40-L42).